src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -23,6 +23,7 @@ public class Http2LoopbackConnection : GenericLoopbackConnection
         private bool _ignoreWindowUpdates;
         public static TimeSpan Timeout => Http2LoopbackServer.Timeout;
         private int _lastStreamId;
+        private bool _shouldSendStatus;
```

Comment:
As I mentioned previously, it seems like a bad idea to store this kind of state on the connection object itself. It can easily lead to the connection getting into an unexpected state for nonobvious reasons.

If you need to change the behavior of some particular method for some reason, it's better to pass in a flag to that specific function.

Reply:
I was thinking about extra parameters but this seems to work quite well. For now, we only have tests with one requests/response. Even when we have more, the logic is simple as we need to set this every time we get new request. Any special complicated cases can be written using protocol specific loopback while keeping GenericLoopback relatively simple.

Reply:
I still think it's a really bad idea.

Is it really that much more work to change this so we don't need to store state on the connection? If so, can you explain to me why?